src/map/roe.cpp

```diff
@@ -43,91 +54,113 @@ void init()
     roeutils::RoeSystem.NotifyThresholds.fill(1);
 
-    if (lua_isnil(L, -1) || !lua_istable(L, -1))
-        return 0;
-
-    // Build caching bitsets from records
+    // Read data from global records table
     lua_pushnil(L);
     while (lua_next(L, -2) != 0)
     {
         // Set Implemented bit.
-        uint32 recordID = static_cast<uint32>(lua_tointeger(L, -2));
+        uint16 recordID = static_cast<uint16>(lua_tointeger(L, -2));
         roeutils::RoeSystem.ImplementedRecords.set(recordID);
 
+        // Register Trigger Handler
+        lua_getfield(L, -1, "trigger");
+        if (!lua_isnil(L, -1))
+        {
+            uint32 trigger = static_cast<uint32>(lua_tointeger(L, -1));
+            if (trigger > 0 && trigger < ROE_NONE)
+                RoeHandlers[trigger].bitmap.set(recordID);
+            else
+                ShowError("ROEUtils: Unknown Record trigger index %d for record %d.", trigger, recordID);
+        }
+        lua_pop(L, 1);
+
         // Set notification threshold
         lua_getfield(L, -1, "notify");
         if (!lua_isnil(L, -1))
             roeutils::RoeSystem.NotifyThresholds[recordID] = static_cast<uint32>((lua_tointeger(L, -1)));
         lua_pop(L, 1);
 
-        // Set repeatability bit
-        lua_getfield(L, -1, "reward");
-        if (!lua_isnil(L, -1))
+        // Set flags
+        lua_getfield(L, -1, "flags");
+        if (!lua_isnil(L, -1) && lua_istable(L, -1))
         {
-            lua_getfield(L, -1, "repeatable");
-            if (lua_toboolean(L, -1))
+            lua_pushnil(L);
+            while (lua_next(L, -2) != 0)
             {
-                roeutils::RoeSystem.RepeatableRecords.set(recordID);
+                std::string flag = lua_tostring(L, -2);
+                if (flag == "daily")
+                {
+                    roeutils::RoeSystem.DailyRecords.set(recordID);
+                    roeutils::RoeSystem.DailyRecordIDs.push_back(recordID);
+                }
+                else if (flag == "timed")
+                {
+                    roeutils::RoeSystem.TimedRecords.set(recordID);
+                }
+                else if (flag == "repeat")
+                {
+                    roeutils::RoeSystem.RepeatableRecords.set(recordID);
+                }
+                else
+                {
+                    ShowError("ROEUtils: Unknown flag %s for record #%d.", flag, recordID);
+                }
+                lua_pop(L, 1);
             }
-            lua_pop(L, 1);
         }
         lua_pop(L, 1);
 
-        lua_pop(L, 1);
+        lua_pop(L, 1); // Pops record entry to prep next loop
     }
     // ShowInfo("\nRoEUtils: %d record entries parsed & available.", RoeBitmaps.ImplementedRecords.count());
     return 0;
 }
 
-bool event(ROE_EVENT eventID, CCharEntity* PChar, RoeDatagramList payload)
+int32 ParseTimedSchedule(lua_State* L)
+{
+    if (lua_isnil(L, -1) || !lua_istable(L, -1))
+            return 0;
```

Comment:
looks like extra indent snuck in here

Reply:
Already caught and fixed in my upcoming chainlink fence commit. :+1: